packages/firestore/src/model/mutation.ts

```diff
@@ -556,15 +634,20 @@ function applyPatchMutationToLocalView(
  */
 function patchDocument(
   mutation: PatchMutation,
-  maybeDoc: MaybeDocument | null
+  maybeDoc: MaybeDocument | null,
+  transformResults?: ProtoValue[]
 ): ObjectValue {
   let data: ObjectValue;
   if (maybeDoc instanceof Document) {
     data = maybeDoc.data();
   } else {
     data = ObjectValue.empty();
   }
-  return patchObject(mutation, data);
+  data = patchObject(mutation, data);
+  if (transformResults) {
```

Comment:
This is never false since you pass in `[]`.

Reply:
Removed the if statement.